code/__DEFINES/jobs.dm

```diff
@@ -342,6 +342,15 @@
 	/datum/job/roguetown/consort,\
 	/datum/job/roguetown/ruler
 
+
+// DUKE, CONSORT, HEIR, PRIEST
+#define CANT_LEAVE_ROLES \
+	/datum/job/roguetown/priest,\
+	/datum/job/roguetown/consort,\
+	/datum/job/roguetown/ruler,\
+	/datum/job/roguetown/prince,\
+	/datum/job/roguetown/hand
```

Comment:
Please define `can_carriage` or a similar variable on `/datum/job` or `/datum/job/roguetown` and check that instead of using this list.